src/dali/plugin/pair_converter/csv_reader/kraken_csv_pricing.py

```diff
@@ -0,0 +1,205 @@
+            for hbar in bars_for_duration:
+                # create keys for every minute starting with longest duration
+                start_epoch: int = int(hbar.timestamp.timestamp())
+                end_epoch: int = int((hbar.timestamp + timedelta(minutes=list(duration_bars.keys())[0])).timestamp())
+                while start_epoch < end_epoch:
+                    timed_bars[datetime.fromtimestamp(start_epoch, timezone.utc)] = hbar
+                    start_epoch += 60
+
+        return timed_bars
+
+    # isolated in order to be mocked
+    def _google_file_to_bytes(self, file_name: str) -> bytes:
+        params: Dict[str, Any] = {
+            _QUERY: f"'{_KRAKEN_FOLDER_ID}' in parents and name = '{file_name}'",
+            _API_KEY: self.__google_api_key,
+        }
+        try:
+            response: Response = self.__session.get(_GOOGLE_APIS_URL, params=params, timeout=self.__TIMEOUT)
+            # {
+            #  "kind": "drive#fileList",
+            #  "incompleteSearch": false,
+            #  "files": [
+            #   {
+            #    "kind": "drive#file",
+            #    "id": "1AAWkwfxJjOvZQKv3c5XOH1ZjoIQMblQt",
+            #    "name": "USDT_OHLCVT.zip",
+            #    "mimeType": "application/zip"
+            #   }
+            #  ]
+            # }
+
+            ## Error Response
+            # {
+            #  'error': {
+            #   'errors': [
+            #    {
+            #     'domain': 'usageLimits',
+            #     'reason': 'accessNotConfigured',
+            #     'message': 'Access Not Configured...',
+            #     'extendedHelp': 'https://console.developers.google.com/apis/api/drive.googleapis.com/overview?project=728279122903'
+            #    }
+            #   ],
+            #   'code': 403,
+            #   'message': 'Access Not Configured...'
+            #  }
+            # }
+            data: Any = response.json()
+
+            error_json: Any = data.get(_ERROR)
+
+            if error_json is not None:
+                errors: Any = error_json[_ERRORS]
+                for error in errors:
+                    if error[_REASON] == _ACCESS_NOT_CONFIGURED:
+                        self.__logger.error(
+                            """
+                            Access not granted to Google Drive API. You must grant authorization to the Google
+                            Drive API for your API key. Follow the link in the message for more details. Message:\n%s
+                        """,
+                            error[_MESSAGE],
+                        )
+                        raise Exception("Google Drive not authorized")
+
+            self.__logger.debug("Retrieved %s from %s", data, response.url)
+
+            params = {_ALT: _MEDIA, _API_KEY: self.__google_api_key, _CONFIRM: 1}  # Bypasses large file warning
+            file_response: Response = self.__session.get(f"{_GOOGLE_APIS_URL}/{data[_FILES][0][_ID]}", params=params, timeout=self.__TIMEOUT)
```

Comment:
Same as above: let's document this in detail or point to other comments documenting this.

Reply:
Ok, I documented this. Basically, the csv reader is d/l the main zipfile that contains 6 csvs, one for each duration.